Cleanup composite's make commands

Others opened the pull request:
### Summary of this Pull Request (PR)
- make now runs make cp automatically
- make init now runs make config automatically if you haven't
- make RUNSCRIPT=my_runscript.sh run is now a thing for running

### Intent for your PR

Choose one (Mandatory):

- [ ] This PR is for a code-review and is intended to get feedback, but not to be pulled yet.
- [x] This PR is mature, and ready to be integrated into the repo.

### Reviewers (Mandatory):
(Specify @<github.com username(s)> of the reviewers. Ex: @user1, @user2)
@gparmer 

### Code Quality

As part of this pull request, I've considered the following:

[Style](https://github.com/gparmer/composite/raw/ppos/doc/style_guide/composite_coding_style.pdf):

- [x] Comments adhere to the Style Guide (SG)
- [x] Spacing adhere's to the SG
- [x] Naming adhere's to the SG
- [x] All other aspects of the SG are adhered to, or exceptions are justified in this pull request
- [ ] I have run the auto formatter on my code before submitting this PR (see doc/auto_formatter.md for instructions)

[Code Craftsmanship](http://www2.seas.gwu.edu/~gparmer/posts/2016-03-07-code-craftsmanship.html):

- [x] I've made an attempt to remove all redundant code
- [x] I've considered ways in which my changes might impact existing code, and cleaned it up
- [x] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [x] I've commented appropriately where code is tricky
- [x] I agree that there is no "throw-away" code, and that code in this PR is of high quality

### Testing

I've tested the code using the following test programs (provide list here):

- making all the things!

phanikishoreg commented:
@Others Just an opinion, I think we should have rust compilation as a conditional compilation for now. 
I'm saying this here because this is a make cleanup commit and it would help. 

I could not get rust to compile fine even after following your instructions on slack. Still get this:
```
***********************************************
*********[ Building Implementations ]**********
***********************************************
|     [--]   Skipping archives
/home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling core v0.0.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore)
   Compiling libc v0.0.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/libc_shim)
   Compiling libc v0.2.30
   Compiling unwind v0.0.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/libunwind)
   Compiling gcc v0.3.53
   Compiling filetime v0.1.10
   Compiling build_helper v0.1.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/build_helper)
   Compiling std v0.0.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd)
   Compiling compiler_builtins v0.0.0 (file:///home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/compiler_builtins_shim)
error: use of deprecated item: gcc::Config has been renamed to gcc::Build
  --> /home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/build.rs:80:20
   |
80 |     let compiler = gcc::Config::new().get_compiler();
   |                    ^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> /home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/build.rs:11:9
   |
11 | #![deny(warnings)]
   |         ^^^^^^^^
   = note: #[deny(deprecated)] implied by #[deny(warnings)]

error: aborting due to previous error

error: Could not compile `std`.
warning: build failed, waiting for other jobs to finish...
warning: use of deprecated item: gcc::Config has been renamed to gcc::Build
    --> /home/phani/.rustup/toolchains/nightly-i686-unknown-linux-gnu/lib/rustlib/src/rust/src/rustc/compiler_builtins_shim/../../libcompiler_builtins/build.rs:4011:24
     |
4011 |         let cfg = &mut gcc::Config::new();
     |                        ^^^^^^^^^^^^^^^^
     |
     = note: #[warn(deprecated)] on by default

error: build failed
error: `"cargo" "build" "--release" "--manifest-path" "/tmp/xargo.sYLjJt0vL1F7/Cargo.toml" "--target" "i686-unknown-composite-musl" "-p" "std"` failed with exit code: Some(101)
note: run with `RUST_BACKTRACE=1` for a backtrace
make[4]: *** [rust] Error 1
|     [CC]   llbooter.c: Compiling

```
Also this part of the compilation takes long and fails for me.

Others commented:
@phanikishoreg There two Rust bugs I am working on fixing, after that it should work for you. I think the solution here is to fix the problems with Rust, rather than just ignoring them. (And if we're going to use more Rust going forward, then we probably want to iron out the kinks now.)

phanikishoreg commented:
I totally agree with you that solution is to fix the problem and I know you've been working hard on that. I'm not suggesting to ignore the problem, absolutely not!

My opinion still is, because it is right now not used by many of us, and not as stable either, it would save a lot of compilation time if we were to make it conditional until we have some initial stability issues ironed out. It's a lot like `stable` vs `nightly` is what I'm talking about.

phanikishoreg commented:
You mentioned you've fixes for those issues, so we can just ignore my comments for now.

gparmer commented on src/Makefile:
```diff
@@ -69,6 +72,9 @@ config-i386: config-gen
 	@echo "i386" > PLATFORM_ID
 
+run:
+	@cd ../transfer && echo "Running" $(RUNSCRIPT) && ./qemu.sh $(RUNSCRIPT)
```
We likely want a default value for `$(RUNSCRIPT)` so that failure to provide it will be more obvious.

gparmer replied:
this requires an update to the documentation in https://github.com/gwsystems/composite/blob/ppos/doc/README.md

Others replied:
@gparmer to be honest, I tried to make it fail with an error when no runscript was provided, but it was beyond my makefoo. How would I do that?

gparmer commented:
Let me know when the documentation is updated and (optionally) RUNSCRIPT has a default value (i.e. "<please provide runscript here>").  I'll pull then.

Others commented:
@gparmer I updated the documentation, and made `make run` alert if no runscript is provided. I also cleaned up some other documentation, but you should probably look at that to make sure my changes aren't wrong / dumb.

gparmer commented:
Looks great to me!  Thanks @Others !!!
